pumps/elasticsearch.go

```diff
@@ -90,57 +141,86 @@ func (e *ElasticsearchPump) WriteData(data []interface{}) error {
 		"prefix": elasticsearchPrefix,
 	}).Info("Writing ", len(data), " records")
 
-	if e.esClient == nil {
+	if e.operator == nil {
 		log.WithFields(logrus.Fields{
 			"prefix": elasticsearchPrefix,
 		}).Debug("Connecting to analytics store")
 		e.connect()
 		e.WriteData(data)
 	} else {
 		if len(data) > 0 {
+			e.operator.processData(data, e.esConf)
+		}
+	}
+	return nil
+}
+
+
+func GetIndexName(esConf *ElasticsearchConf) string {
+	var indexName = esConf.IndexName
+
+	if esConf.RollingIndex {
+		var currentTime = time.Now()
+		//This formats the date to be YYYY.MM.DD but Golang makes you use a specific date for its date formatting
+		indexName = indexName + "-" + currentTime.Format("2006.01.02")
+	}
+	return indexName
+}
+
+func GetMapping(datum interface{}, extendedStatistics bool) map[string]interface{} {
+	var record, _ = datum.(analytics.AnalyticsRecord)
+
+	mapping := map[string]interface{}{
+		"@timestamp":      record.TimeStamp,
+		"http_method":     record.Method,
+		"request_uri":     record.Path,
+		"response_code":   record.ResponseCode,
+		"ip_address":      record.IPAddress,
+		"api_key":         record.APIKey,
+		"api_version":     record.APIVersion,
+		"api_name":        record.APIName,
+		"api_id":          record.APIID,
+		"org_id":          record.OrgID,
+		"oauth_id":        record.OauthID,
+		"request_time_ms": record.RequestTime,
+	}
+
+	if extendedStatistics {
+		mapping["raw_request"] = record.RawRequest
+		mapping["raw_response"] = record.RawResponse
+		mapping["user_agent"] = record.UserAgent
+	}
+	return mapping
+}
+
+func(e Elasticsearch3Operator) processData(data []interface{}, esConf *ElasticsearchConf) error {
+	var index = e.esClient.Index().Index(GetIndexName(esConf))
```

Comment:
You like `var` a lot - could you use the shorter `:=` instead? Unless the rest of the code uses `var` more frequently, which I would assume it doesn't.